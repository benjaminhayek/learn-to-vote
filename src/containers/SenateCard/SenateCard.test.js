import React from 'react';
import { SenateCard } from './index.js';
import { mapStateToProps, mapDispatchToProps } from './index.js';
import { shallow, mount } from 'enzyme';

describe('SenateCard', () => {
  let wrapper;
  let mockFetch;
  let mockParty;
  let senator;
  let toggleSelected;

  beforeEach(() => {
    mockFetch = jest.fn();
    mockParty = 'D';
    toggleSelected = jest.fn()
    senator = {name: 'name', party: 'D', title: 'title'}
    wrapper = shallow(
      <SenateCard senator={senator}/>
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call toggleSelect on click', () => {
    wrapper = mount(<SenateCard senator={senator} toggleSelected={toggleSelected}/>)
    const spy = jest.spyOn(wrapper.instance(), 'toggleSelect');
    wrapper.instance().forceUpdate();
    wrapper.find('.dem').simulate('click')
    expect(spy).toHaveBeenCalled()
  })
})

describe('mapStateToProps', () => {
  it('should have access to senators, congressmen and loading', () => {
    const mockStore = {
      selected: 'undefined',
    }
    const expected = {...mockStore};
    const result = mapStateToProps(mockStore);
    expect(result).toEqual(expected);
  });
})

describe('mapDispatchToProps', () => {
  it('should call dispatch when fetchCongress is invoked', () => {
    const mockDispatch = jest.fn();
    const mappedProps = mapDispatchToProps(mockDispatch);
    mappedProps.toggleSelected({name: 'Political Guy'});
    expect(mockDispatch).toHaveBeenCalled();
  });
})