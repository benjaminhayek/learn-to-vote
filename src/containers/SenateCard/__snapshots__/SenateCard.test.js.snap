// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SenateCard should match the snapshot 1`] = `
<div
  className="dem"
  onClick={[Function]}
>
  <img
    alt="apple"
    className="apple"
    src="apple.svg"
  />
  <h1
    className="name"
  >
    name
  </h1>
  <h2
    className="title"
  >
    title
  </h2>
  <h2
    className="party"
  >
    D
  </h2>
  <h2
    className="election-date"
  >
    Next Election: 
  </h2>
</div>
`;
